FormWidget.js

```diff
@@ -131,7 +143,17 @@ define([
 
 			// Set tabIndex etc. for all tabbable nodes.
 			// To keep things simple, if anything has changed then reapply all the properties.
-			if ("tabStops" in oldValues || "tabIndex" in oldValues || "disabled" in oldValues || "alt" in oldValues) {
+			if ("tabStops" in oldValues || "tabIndex" in oldValues || "disabled" in oldValues
+				|| "alt" in oldValues || "required" in oldValues) {
+			
+				var inputIdUnset = true;
+				var inputId;
+				if ("id" in oldValues) {
+					if (this.id) {
+						inputId = this.id + "-input";
+					}
+				}
```

Comment:
The code block from line 148 - 188 should presumably run whenever `id` changes, but it doesn't necessarily do so, because it only runs if tabStops, tabIndex, disabled, alt, or required has changed.

You can also collapse those two `if()` into a single one, and likewise on line 180 below.